src/modules/land_detector/FixedwingLandDetector.h

```diff
@@ -57,48 +57,46 @@ class FixedwingLandDetector : public LandDetector
 	FixedwingLandDetector();
 
 protected:
-	virtual void _initialize_topics() override;
+	void _initialize_topics() override;
+	void _update_params() override;
+	void _update_topics() override;
+	bool _get_landed_state() override;
+	bool _get_maybe_landed_state() override;
+	bool _get_ground_contact_state() override;
+	bool _get_freefall_state() override;
+	float _get_max_altitude() override;
 
-	virtual void _update_params() override;
-
-	virtual void _update_topics() override;
-
-	virtual bool _get_landed_state() override;
-
-	virtual bool _get_maybe_landed_state() override;
-
-	virtual bool _get_ground_contact_state() override;
```

Comment:
I thought about multiple inheritance as well, but I was not able to come up with a good implementation for the diamond problem, since the switch is guided by the flag rotary_wing.